dev-lang/icon/files/icon-9.5.1-musl.patch

```diff
@@ -0,0 +1,23 @@
+--- icon-v951/ipl/cfuncs/fpoll.c	2020-04-12 16:46:27.195467895 -0600
++++ icon-v951-fix/ipl/cfuncs/fpoll.c	2020-05-06 18:36:54.220286782 -0600
+@@ -33,6 +33,10 @@
+ #include <sys/types.h>
+ #include <sys/time.h>
+ 
++#if defined(_GENTOO_MUSL_FIX)	/* MUSL library */
++# include <stdio_ext.h>
++#endif
++
+ #include "icall.h"
+ 
+ int fpoll(int argc, descriptor *argv)	/*: await data from file */
+@@ -63,6 +67,9 @@ int fpoll(int argc, descriptor *argv)	/*
+ #if defined(__GLIBC__)                  /* New GCC library */
+    if (f->_IO_read_ptr < f->_IO_read_end)
+       RetArg(1);
++#elif defined(_GENTOO_MUSL_FIX)                         /* MUSL library */
```

Comment:
You should check for linux and not glibc this prevents from having to have a custom define

Reply:
No.. the patch file is wrong that I will need to fix.

When I created the patch file I used the wrong "old" directory.  
I will correct it.

As for the GLIBC check, that is from upstream.  The source is built on a wide range of environments and there are other platforms than Linux that use GLIBC.

Regardless.. I do need to fix the patch file however.

> You should check for linux and not glibc this prevents from having to have a custom define

Reply:
Scratch that last comment.. the file is correct (I was misviewing the "+" in the message).

Regardless.. the test for glibc is from upstream